src/client/app/components/csv/ReadingsCSVUploadComponent.tsx

```diff
@@ -26,315 +33,477 @@ function range(lower: number, upper: number): number[] {
+		<Container>
+			<TooltipHelpComponent page='help.csv.header' />
+			<Form onSubmit={handleSubmit}>
+				<Row className="justify-content-md-center">
+					<Col md='auto'>
+						<h2>
+							{translate('csv.upload.readings')}
+							<div style={tooltipStyle}>
+								<TooltipMarkerComponent page='help.csv.header' helpTextId={tooltipStyle.tooltipReadings} />
+							</div>
+						</h2>
+					</Col>
+				</Row>
+				<Row className='justify-content-md-center'>
+					<Col md='auto'>
+						<Label for='meterName'>
+							<div className='pb-1'>
+								{translate('name')}
+							</div>
+							<Input
+								id='meterName'
+								name='meterName'
+								type='select'
+								value={selectedMeter?.name || ''}
+								onChange={handleSelectedMeterChange}>
+								<option value='default' key='-1'>Select a Meter</option>
+								{Array.from(visibleMeters).map(meter => {
+									return (<option value={meter.name} key={meter.id}>{meter.name}</option>);
+								})}
 							</Input>
-						</Col>
-					</FormGroup>
-					<FormFileUploaderComponent formText='csv.upload.readings' reference={this.fileInput} required labelStyle={titleStyle} />
-					<FormGroup>
-						<Label style={titleStyle}>
-							<FormattedMessage id='csv.readings.section.cumulative.data' />
 						</Label>
-						<Col sm={8}>
-							<FormGroup>
-								<Label style={titleStyle}>
-									<FormattedMessage id='csv.readings.param.cumulative' />
-								</Label>
-								<Col sm={12}>
-									<Input type='select' name='cumulative' onChange={this.handleSetCumulative}>
-										<option value={BooleanMeterTypes.meter}> {translate('BooleanMeterTypes.meter')} </option>
-										<option value={BooleanMeterTypes.true}> {translate('BooleanMeterTypes.true')} </option>
-										<option value={BooleanMeterTypes.false}> {translate('BooleanMeterTypes.false')} </option>
-									</Input>
+						<br />
+						<CreateMeterModalComponent onCreateMeter={handleCreateMeter}/>
+						<br /><br />
+						<FormGroup>
+							<Label for='timeSort'>
+								<div className='pb-1'>
+									{translate('meter.timeSort')}
+								</div>
+								<Input
+									type='select'
+									id='timeSort'
+									name='timeSort'
+									value={selectedMeter?.timeSort || ''}
+									onChange={handleChange}
+								>
+									<option value={TimeSortTypes.meter}> {translate('TimeSortTypes.meter')} </option>
+									<option value={TimeSortTypes.increasing}> {translate('TimeSortTypes.increasing')} </option>
+									<option value={TimeSortTypes.decreasing}> {translate('TimeSortTypes.decreasing')} </option>
+								</Input>
+							</Label>
+						</FormGroup>
+						<FormGroup>
+							<Label for='duplications'>
+								<div className='pb-1'>
+									{translate('meter.readingDuplication')}
+								</div>
+								<Input type='select' id='duplications' name='duplications' value={selectedMeter?.readingDuplication || ''} onChange={handleChange}>
+									{range(1, 10).map(i => (
```

Comment:
I like the way this is done. The recent change to the meter create/edit pages lists each one. Are you willing to roll this to those pages? Maybe put the range functions into a general util directory.